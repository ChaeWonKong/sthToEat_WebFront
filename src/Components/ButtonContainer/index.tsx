import React from "react";
import { Button, ButtonGroup } from "react-bootstrap";
import "./buttonContainer.scss";

const ButtonContainer: React.FC = () => {
  return (
    <div className="wrapper">
      <ButtonGroup aria-label="Basic example">
        <Button variant="danger">DISLIKE</Button>
        <Button variant="primary">LIKE</Button>
      </ButtonGroup>
    </div>
  );
};

export default ButtonContainer;
