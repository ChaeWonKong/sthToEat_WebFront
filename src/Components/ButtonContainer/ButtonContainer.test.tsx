import React from "react";
import { shallow } from "enzyme";
import ButtonContainer from "./index";

describe("ButtonContainer", () => {
  const buttonContainer = shallow(<ButtonContainer />);
  it("renders correctly", () => {
    expect(buttonContainer).toMatchSnapshot();
  });

  it("has `wrapper` class", () => {
    expect(buttonContainer.find(".wrapper").exists()).toBe(true);
  });

  // Button Group
  describe("ButtonGroup", () => {
    let buttonGroup = buttonContainer.find("ButtonGroup");

    it("renders correctly", () => {
      expect(buttonGroup.exists()).toBe(true);
    });

    it("has two children", () => {
      expect(buttonGroup.children("Button").length).toEqual(2);
    });
  });

  it("has ButtonGroup", () => {});

  it("has two buttons", () => {
    expect(buttonContainer.find("Button").length).toEqual(2);
  });
});
