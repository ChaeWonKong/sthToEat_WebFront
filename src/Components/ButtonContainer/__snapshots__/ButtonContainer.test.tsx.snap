// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonContainer renders correctly 1`] = `
<div
  className="wrapper"
>
  <ButtonGroup
    aria-label="Basic example"
    role="group"
    toggle={false}
    vertical={false}
  >
    <Button
      active={false}
      disabled={false}
      type="button"
      variant="danger"
    >
      DISLIKE
    </Button>
    <Button
      active={false}
      disabled={false}
      type="button"
      variant="primary"
    >
      LIKE
    </Button>
  </ButtonGroup>
</div>
`;
